configuration/config2/SchemaItem.hpp

```diff
@@ -73,6 +77,11 @@ class DECL_EXPORT SchemaItem : public std::enable_shared_from_this<SchemaItem>
         void setHidden(bool hidden) { m_hidden = hidden; }
         bool isHidden() const { return m_hidden; }
 
+        void setParent(const std::shared_ptr<SchemaItem> &parent) { m_pParent = parent; }
+        std::shared_ptr<const SchemaItem> findSchemaRoot() const;
```

Comment:
I understand the implementation "walks" the parent nodes, but conceptually this doesn't seem like a "find" function,.. more like a "getSchemaRoot()" or similiar name.

Reply:
True, it is a get, find was carried over because it was written while I was implementing the find functionality. I changed it to getSchemaRoot. Similar change made to finding the root node in the environment.